src/test/java/hudson/plugins/copyartifact/CopyArtifactWorkflowTest.java

```diff
@@ -146,4 +156,173 @@ private WorkflowJob createWorkflow(String name, String script) throws IOExceptio
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: lastCompleted()));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testParameterizedBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: buildParameter('<StatusBuildSelector/>')));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testPermalinkBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: permalink('lastStableBuild')));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testSavedBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0)).keepLog();
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: latestSavedBuild()));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testSpecificBuildSelector() throws Exception {
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: specific(\"${build('copiee').number}\")));"
```

Comment:
or

```groovy
selector: specific(build('copiee').id)
```

(BTW `SpecificBuildSelector` should perhaps use `Job.getBuild(String)` rather than looking for arguments that seem to be numbers.)

Reply:
Created https://issues.jenkins-ci.org/browse/JENKINS-47074